datastore/bolt/event_delivery.go

```diff
@@ -0,0 +1,323 @@
+		} `json:"event_metadata"`
+	}
+
+	err := e.db.View(func(tx *bbolt.Tx) error {
+
+		var eid eid
+		c := tx.Bucket([]byte(eventDeliveryBucketName)).Cursor()
+
+		var deliverySlice [][]byte
+
+		// seek all event deliveries
+		for k, v := c.First(); k != nil; k, v = c.Next() {
+			fmt.Printf("key=%s, value=%s\n", k, v)
+
+			err := json.Unmarshal(v, &eid)
+			if err != nil {
+				return err
+			}
+
+			if eid.EventMetadata.UID != eventID {
+				continue
+			}
+
+			deliverySlice = append(deliverySlice, v)
+		}
+
+		deliveries = make([]datastore.EventDelivery, len(deliverySlice))
+		for i, buf := range deliverySlice {
+			var delivery datastore.EventDelivery
+			err := json.Unmarshal(buf, &delivery)
+			if err != nil {
+				return err
+			}
+
+			deliveries[i] = delivery
+		}
+
+		return nil
+	})
+
+	return deliveries, err
+}
+
+func (e *eventDeliveryRepo) UpdateStatusOfEventDelivery(ctx context.Context, delivery datastore.EventDelivery, status datastore.EventDeliveryStatus) error {
+	delivery.Status = status
+	delivery.UpdatedAt = primitive.NewDateTimeFromTime(time.Now())
+
+	return createUpdateEventDelivery(e.db, &delivery)
+}
+
+func (e *eventDeliveryRepo) UpdateEventDeliveryWithAttempt(ctx context.Context, delivery datastore.EventDelivery, attempt datastore.DeliveryAttempt) error {
+	delivery.DeliveryAttempts = append(delivery.DeliveryAttempts, attempt)
+
+	return createUpdateEventDelivery(e.db, &delivery)
+}
+
+func (e *eventDeliveryRepo) LoadEventDeliveriesPaged(ctx context.Context, groupID, appID, eventID string, status []datastore.EventDeliveryStatus, searchParams datastore.SearchParams, pageable datastore.Pageable) ([]datastore.EventDelivery, datastore.PaginationData, error) {
+	f := &filter{
+		groupID:      groupID,
+		appID:        appID,
+		eventID:      eventID,
+		status:       status,
+		searchParams: searchParams,
+
+		hasAppFilter:       !util.IsStringEmpty(appID),
+		hasGroupFilter:     !util.IsStringEmpty(groupID),
+		hasEventFilter:     !util.IsStringEmpty(eventID),
+		hasStatusFilter:    len(status) > 0,
+		hasStartDateFilter: searchParams.CreatedAtStart > 0,
+		hasEndDateFilter:   searchParams.CreatedAtEnd > 0,
+	}
+
+	if pageable.Page < 1 {
+		pageable.Page = 1
+	}
+	if pageable.PerPage < 1 {
+		pageable.PerPage = 10
+	}
+
+	prevPage := pageable.Page - 1
+	lowerBound := pageable.PerPage * prevPage
+	upperBound := pageable.PerPage * pageable.Page
+
+	var deliveries []datastore.EventDelivery
+	var pg datastore.PaginationData
+	err := e.db.View(func(tx *bbolt.Tx) error {
+
+		b := tx.Bucket([]byte(eventDeliveryBucketName))
+		c := b.Cursor()
+
+		i := 0
+		// seek all event deliveries
+		for k, v := c.First(); k != nil; k, v = c.Next() {
+			if i >= lowerBound && i < upperBound {
+				var d datastore.EventDelivery
+				err := json.Unmarshal(v, &d)
+				if err != nil {
+					return err
+				}
+
+				if !e.filterEventDelivery(f, &d) {
+					continue
+				}
+
+				deliveries = append(deliveries, d)
+			}
+			i++
```

Comment:
We should only increment when we are within the bounds
```suggestion
				deliveries = append(deliveries, d)
				i++
			}
```